eng/pipelines/templates/stages/archetype-sdk-client.yml

```diff
@@ -7,6 +7,7 @@ stages:
     jobs:
     - template: ../jobs/archetype-sdk-client.yml
       parameters:
+        ServiceToBuild: ${{parameters.ServiceDirectory}}
```

Comment:
I don't think this is going to do what you want. I think you want something like:

```
 ServiceToBuild: ${{ coalesce(parameters.ServiceToBuild, parameters.ServiceDirectory) }}
```
see more information about coalesce at https://docs.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops#coalesce